src/rules/package-lint-dependencies-conform.ts

```diff
@@ -0,0 +1,88 @@
+import { buildRule } from './build-rule';
+import { PackageManifestSchema, RuleName } from './types';
+import type { RuleExecutionFailure } from '../execute-rules';
+
+export default buildRule({
+  name: RuleName.PackageLintDependenciesConform,
+  description:
+    'Do the lint-related `devDependencies` in `package.json` conform?',
+  dependencies: [RuleName.RequireValidPackageManifest],
+  execute: async ({ project, template, pass, fail }) => {
+    const entryPath = 'package.json';
+
+    const templateManifest = await template.fs.readJsonFileAs(
+      entryPath,
+      PackageManifestSchema,
+    );
+
+    const projectManifest = await project.fs.readJsonFileAs(
+      entryPath,
+      PackageManifestSchema,
+    );
+
+    const failures: RuleExecutionFailure[] = await lintPackageConform(
+      templateManifest.devDependencies,
+      projectManifest.devDependencies,
+    );
+
+    return failures.length === 0 ? pass() : fail(failures);
+  },
+});
+
+/**
+ * Validates whether target project has all the required lint packages with versions matching with template project.
+ *
+ * @param templateDependencies - The record of lint package name and version from template.
+ * @param projectDependencies - The record of lint package name and version from project.
+ */
+async function lintPackageConform(
+  templateDependencies: Record<string, string>,
+  projectDependencies: Record<string, string>,
+): Promise<RuleExecutionFailure[]> {
+  const templateLintPackageNames = await getTemplateLintPackageNames(
+    templateDependencies,
+  );
+  const failures: RuleExecutionFailure[] = [];
+  for (const lintPackage of templateLintPackageNames) {
+    const projectPackageVersion = projectDependencies[lintPackage];
+    const templatePackageVersion = templateDependencies[lintPackage] as string;
+    if (!projectPackageVersion) {
+      failures.push({
+        message: `\`package.json\` should list \`"${lintPackage}": "${templatePackageVersion}"\` in \`devDependencies\`, but does not.`,
+      });
+
+      continue;
+    }
+
+    if (projectPackageVersion !== templatePackageVersion) {
+      failures.push({
+        message: `${lintPackage} version is "${projectPackageVersion}", when it should be "${templatePackageVersion}".`,
```

Comment:
This might have gotten lost in the previous discussions, but what do you think about making this failure message more consistent with the previous  message?

``` suggestion
        message: `\`package.json\` lists "${lintPackage}" in \`devDependencies\` with version "${projectPackageVersion}", but the version should be "${templatePackageVersion}".`,

Reply:
Ah, okay, that's fair. Perhaps we can go through and review all of the error messages later. In the  meantime, do we want to at least mention that the package is in the `devDependencies` field? What I'm worried about is that if the user sees this message, they're not going to know what to change exactly to fix the lint violation.

Reply:
My thought was that the user might not read the header, but only the failure message. But you're right. The information is all there already. This is fine and we can revisit later if it becomes an issue.